pkg/apis/extensions/types.go

```diff
@@ -228,6 +228,14 @@ type DeploymentSpec struct {
 	// Value of this key is hash of DeploymentSpec.PodTemplateSpec.
 	// No label is added if this is set to empty string.
 	UniqueLabelKey string `json:"uniqueLabelKey,omitempty"`
+
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	Rollback *RollbackConfig `json:"rollback,omitempty"`
+}
+
+type RollbackConfig struct {
```

Comment:
The subresource needs another top-level object with at least TypeMeta and annotations, which can inline RollbackConfig.
```
type DeploymentRollback struct {
  TypeMeta
  Annotations map[string]string
  RollbackConfig
}
```